context/http.go

```diff
@@ -104,18 +104,6 @@ func GetRequestID(ctx context.Context) string {
 // WithResponseWriter returns a new context and response writer that makes
 // interesting response statistics available within the context.
 func WithResponseWriter(ctx context.Context, w http.ResponseWriter) (context.Context, http.ResponseWriter) {
-	if closeNotifier, ok := w.(http.CloseNotifier); ok {
```

Comment:
Can you provide a little more context in the commit message as to why the notifier is being removed?

Reply:
`staticchecks` was failing since `http.CloseNotifier` is deprecated. So we either remove the notifier and rely on `Request.Context` instead or remove the checks. I opted for the former